### PR TITLE
(CDAP-11846) + (CDAP-7651) Fix Hive on Spark

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-RUT1387

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java [hunk 1]
```diff
@@ -203,7 +202,6 @@ protected BaseHiveExploreService(TransactionSystemClient txClient, DatasetFramew
                                    AuthenticationContext authenticationContext) {
     this.cConf = cConf;
     this.hConf = hConf;
-    this.hiveConf = createHiveConf();
```
Comment (anwar6953) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: this commit was causing failures?

Comment (chtyim) [reply cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: The Hive on Spark failure? No, it is mainly caused by the hack through the `ResourcesClassLoader`, which get removed in this PR.

**Comment by anwar6953** [comment]
Kicked off integration tests against a CDH 5.5 cluster: https://builds.cask.co/browse/CDAP-ITM19-34

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java (hunk 1 above)
Comment (anwar6953) [reply cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: I mean - why'd you revert this caching change, then?

**Review by anwar6953 — APPROVED** [review]

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java (hunk 1 above)
Comment (chtyim) [reply cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: It causes some unit-test failure, apparently the hiveConf can't be shared between sessions. I haven't dig deep into why, but instead reverting to a new instance

**Comment by anwar6953** [comment]
I believe this PR also fixed https://issues.cask.co/browse/CDAP-12415.

**Pull request merged**